blocks/library/image/editor.scss

```diff
@@ -87,3 +87,9 @@
 		margin-right: auto;
 	}
 }
+
+.editor-block-list__block[data-type="core/image"][data-align="right"] {
+	.wp-block-image {
+		float: right;
+	}
+}
```

Comment:
If I understand properly, this fix PR is meant for the frontend, is this really necessary? we just merged https://github.com/WordPress/gutenberg/pull/5142 that refactors how float work where the idea was to make floats work in a generic way in the editor not requiring specific handling per block.

Can you explain why do we need this specific handling? right now the block nodes (`wp-block-image`) are not floated in master but their container is in a generic way to avoid having to do it in each block.

Reply:
On further repo reading, I ran into the same bug as #3945, but it may require more attention to properly address the behaviour that causes #3944.